MoviesApi/MoviesApi/Controllers/RolesController.cs

```diff
@@ -0,0 +1,121 @@
+﻿using AutoMapper;
+using ETMovies.Models;
+using ETMovies.Service;
+using Microsoft.AspNetCore.Mvc;
+using MoviesApi.ModelsDTO;
+
+namespace MoviesApi.Controllers
+{
+    [Route("api/[controller]/[action]")]
+    [ApiController]
+    public class RolesController : Controller
+    {
+
+        private RoleService service;
+        ILogger<RolesController> logger;
+        private IMapper mapper;
+        public RolesController(RoleService roleService, ILogger<RolesController> logger, IMapper mapper)
+        {
+            service = roleService;
+            this.logger = logger;
+            this.mapper = mapper;
+        }
+
+        [HttpGet]
+        public IEnumerable<RoleReadDto> GetRoles(int page = 1, int pageSize = 10)
+        {
+            logger.LogInformation(" ------ ");
+            logger.LogInformation("GetRoles WAS CALLED");
+            logger.LogInformation(" ------ ");
+            var roles = service.GetRoles();
+            if(roles == null)
+            {
+                logger.LogInformation("No roles found");
+                return null;
+            }
+            var rolesDto = mapper.Map<List<RoleReadDto>>(roles);
+            logger.LogInformation(" ------ ");
+            logger.LogInformation("Returning roles");
+            logger.LogInformation(" ------ ");
+            return rolesDto.Skip((page - 1) * pageSize).Take(pageSize).ToList();
+
+        }
+
+        [HttpGet("{id}")]
+        public RoleReadDto GetRole(int id)
+        {
+            logger.LogInformation(" ------ ");
+            logger.LogInformation("GetRole{id} WAS CALLED");
+            logger.LogInformation(" ------ ");
+            var role = service.GetRole(id);
+            if (role == null)
+            {
+                logger.LogInformation("No role Found");
+                return null;
+            }
+            var roleDto = mapper.Map<RoleReadDto>(role);
+            logger.LogInformation(" ------ ");
+            logger.LogInformation("Returning role");
+            logger.LogInformation(" ------ ");
+            return roleDto;
+
+        }
+
+        [HttpPost]
+        public void Post(string name, int personID, int movieID)
+        {
+            logger.LogInformation(" ------ ");
+            logger.LogInformation("Post WAS CALLED");
+            logger.LogInformation(" ------ ");
+            var roleDto = new RoleReadDto(name, personID, movieID);
```

Comment:
you should already accept a DTO in the POST, if you do that, you won't need to create objects everytime